geode-core/src/main/java/org/apache/geode/security/ResourcePermission.java

```diff
@@ -68,6 +77,20 @@ public ResourcePermission() {
     this(Resource.NULL, Operation.NULL, ALL, ALL);
   }
 
+  public ResourcePermission(String resource, String operation) {
+    this(resource, operation, ALL, ALL);
+  }
+
+  public ResourcePermission(String resource, String operation, String target) {
+    this(resource, operation, target, ALL);
+  }
+
+  public ResourcePermission(String resource, String operation, String target, String key) {
+    this((resource == null) ? Resource.NULL : Resource.valueOf(resource.toUpperCase()),
```

Comment:
This is causing the `String`-based to go through the `Enum`-based methods.  I think we want to do that the other way around since the `String` methods are more general.  (For example, if someone wants to check for some custom security target like "CLUSTER:READ:LDAP" this will currently throw an exception.)

Reply:
Only resource/operation will be turned into enum (because internally we save the resource/operation in enum form, not string form), but target and key are in String form. So new ResourePermission("cluster", "read", "ldap") won't throw an exception.

Reply:
I think it seems more extensible to use the Strings in our internal representation rather than the Enums.  (So that a new resource or operation could be introduced without a breaking change.) Is there a reason that makes you prefer the Enums as our internal representation?

Reply:
Not particular reason, just thought we will cross that bridge when we get there. Currently we are not allowing other resource/operations. If we make that change now, we will be allowing that which may not be the intention of this changeset.

Reply:
or do we want to jump ahead and allow other resource/operation with this change and deprecate the ENUM form (along with getResource() and getOperation() call)?